vertx-template-engines/vertx-web-templ-freemarker/src/main/java/io/vertx/ext/web/templ/freemarker/impl/FreeMarkerTemplateEngineImpl.java

```diff
@@ -53,29 +54,22 @@ public FreeMarkerTemplateEngine setExtension(String extension) {
     return this;
   }
 
-  @Override
-  public FreeMarkerTemplateEngine setMaxCacheSize(int maxCacheSize) {
-    this.cache.setMaxSize(maxCacheSize);
-    return this;
-  }
-
   @Override
   public void render(Map<String, Object> context, String templateFile, Handler<AsyncResult<Buffer>> handler) {
     try {
-      Template template = isCachingEnabled() ? cache.get(templateFile) : null;
+      String src = adjustLocation(templateFile);
+      TemplateHolder<Template> template = getTemplate(src);
```

Comment:
Can you extract this in a template method inside `CachingTemplateEngine`? It's always the same logic it seems, so we could just provide a lambda to create the template for each specific engine.